com.unity.netcode.gameobjects/Runtime/NetworkVariable/Collections/NetworkSet.cs

```diff
@@ -1,18 +1,19 @@
 #if !NET35
-using System.Collections;
-using System.Collections.Generic;
+using System;
 using System.IO;
+using Unity.Collections;
 
 namespace Unity.Netcode
 {
     /// <summary>
     /// Event based NetworkVariable container for syncing Sets
     /// </summary>
     /// <typeparam name="T">The type for the set</typeparam>
-    public class NetworkSet<T> : NetworkVariableBase, ISet<T> where T : unmanaged
+    public class NetworkSet<T> : NetworkVariableBase where T : unmanaged, IEquatable<T>
```

Comment:
I assume `where T : unmanaged, IEquatable<T>` part is enforced because of constraints coming from underlying data structure's (NativeHashMap or NativeHashSet), right?

Reply:
Yes, Native Collections require T to be a struct (or unmanaged depending on the collection), though they do not require them to be IEquatable